nomenclature/__init__.py

```diff
@@ -126,3 +161,56 @@ def _validate_directional(x):
     """Utility function to check whether region-to-region code is valid"""
     x = x.split('>')
     return len(x) == 2 and all([i in regions for i in x])
+
+
+def _validate_subannual_dt(x):
+    """Utility function to separate and validate datetime format"""
+    valid_dt, invalid_tz, invalid = [], False, set()
+    for (y, s) in x:
+        try:  # casting to Central European datetime
+            valid_dt.append(datetime.strptime(f'{y}-{s}', '%Y-%m-%d %H:%M%z'))
+        except ValueError:
+            try:  # casting to UTC datetime
+                datetime.strptime(f'{y}-{s}', '%Y-%m-%d %H:%M')
+                invalid_tz = True
+            except ValueError:  # if casting to datetime fails, return invalid
+                invalid.add(s)
+    return valid_dt, invalid_tz, list(invalid)
+    
+    
+def swap_time_for_subannual(df):
+    """Convert an IamDataFrame with `datetime` domain to `year + subannual`"""
+    if df.time_col != 'time':
+        raise ValueError('The IamDataFrame does not have `datetime` domain!')
+
+    _data = df.data
+    _data['year'] = [x.year for x in _data.time]
+    _data['subannual'] = [x.strftime('%m-%d %H:%M%z') for x in _data.time]
+    _data.drop(columns='time', inplace=True)
+
+    return IamDataFrame(_data)
+
+
+def validate_time_dt(x):
```

Comment:
suggest to rename to "internal" function by adding `_`
```suggestion
def _validate_time_dt(x):
```